service/src/main/java/org/cbioportal/service/impl/UuidDataAccessTokenServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+            dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
+        } 
+    }
+
+    @Override
+    public void revokeDataAccessToken(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
+    }
+    
+    @Override
+    public String getUsername(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken.getUsername();
+    }
+    
+    @Override
+    public Date getExpiration(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken.getExpiration();
+    }
+   
+    @Override
+    public Boolean isValid(String dataAccessToken) {
+        DataAccessToken storedDataAccessToken = null;
+        try {
+            storedDataAccessToken = dataAccessTokenRepository.getDataAccessToken(dataAccessToken);
+        } catch (Exception e) {
+            log.error("Error retrieving data access token, " + dataAccessToken + " from token store");
+            return Boolean.FALSE;
+        }
+        Calendar calendar = Calendar.getInstance();
+        Date currentDate = calendar.getTime();
+        if (storedDataAccessToken.getExpiration().before(currentDate)) {
+            return Boolean.FALSE;
+        } 
+        return Boolean.TRUE;
+    }
+
+    private int getNumberOfTokensForUsername(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        return allDataAccessTokens.size();
+    }
+
+    // revokes oldest token in token management system for a user
+    private void revokeOldestDataAccessTokenForUsername(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        DataAccessToken oldestDataAccessToken = getOldestDataAccessToken(allDataAccessTokens);
+        dataAccessTokenRepository.removeDataAccessToken(oldestDataAccessToken.getToken());
+    }
+    
+    // helper function for getting the newest data access token out of a list of data access tokens
+    private DataAccessToken getNewestDataAccessToken(List<DataAccessToken> dataAccessTokens) {
+        DataAccessToken newestDataAccessToken = null;
+        for (DataAccessToken dataAccessToken : dataAccessTokens) {
+            newestDataAccessToken = (newestDataAccessToken != null) ? (getNewerDataAccessToken(newestDataAccessToken, dataAccessToken)) : (dataAccessToken);
+        }
+        return newestDataAccessToken;
+    }
+
+    // helper function for comparing two data access tokens and returning the newer one
+    private DataAccessToken getNewerDataAccessToken(DataAccessToken newestDataAccessToken, DataAccessToken dataAccessToken) {
```

Comment:
rather than a method here to compare two tokens, this could be added to the model class so that other places the code would be able to compare to tokens.
public boolean expiresSoonerThan(DataAccessToken otherToken);